core/engine/src/bytecompiler/declarations.rs

```diff
@@ -24,6 +25,361 @@ use boa_ast::operations::annex_b_function_declarations_names;
 
 use super::{Operand, ToJsString};
 
+/// `GlobalDeclarationInstantiation ( script, env )`
+///
+/// This diverges from the specification by separating the context from the compilation process.
+/// Many steps are skiped that are done during bytecode compilation.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-globaldeclarationinstantiation
+#[cfg(not(feature = "annex-b"))]
+#[allow(clippy::unnecessary_wraps)]
+#[allow(clippy::ptr_arg)]
+pub(crate) fn global_declaration_instantiation_context(
+    _annex_b_function_names: &mut Vec<Identifier>,
+    _script: &Script,
+    _env: &Rc<CompileTimeEnvironment>,
+    _context: &mut Context,
+) -> JsResult<()> {
+    Ok(())
+}
+
+/// `GlobalDeclarationInstantiation ( script, env )`
+///
+/// This diverges from the specification by separating the context from the compilation process.
+/// Many steps are skiped that are done during bytecode compilation.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-globaldeclarationinstantiation
+#[cfg(feature = "annex-b")]
+pub(crate) fn global_declaration_instantiation_context(
```

Comment:
Thought: Potentially move these out of `bytecompiler` to `script`

Now that these are separated, is there any need for these functions to still live in this module (especially if the bytecompiler is eventually moved to it's own crate).

Reply:
Yup, we can put them in separate locations, the only reason I left them there was because the context and non-context variants are close to each other.